### ISSUE TITLE
Exception trying to coerce int and ulongs

### ISSUE THREAD

**Issue by mikechristiansenvae** [scriban/scriban#395]
Steps to replicate:

Imported object:

```c#
public class MyObject 
{
    public ulong MyValue { get; set; }
}
```

Template:

```
if my_object.my_value > 0
  "Larger than zero"
end
```

Stack Trace:

```
Scriban.Syntax.ScriptRuntimeException : <input>(195,19) : error : Unable to convert type `ulong` to int
  ----> System.OverflowException : Value was either too large or too small for an Int32.
   at Scriban.TemplateContext.ToInt(SourceSpan span, Object value)
   at Scriban.TemplateContext.ToObject(SourceSpan span, Object value, Type destinationType)
   at Scriban.Syntax.ScriptBinaryExpression.CalculateOthers(TemplateContext context, SourceSpan span, ScriptBinaryOperator op, SourceSpan leftSpan, Object leftValue, SourceSpan rightSpan, Object rightValue)
   at Scriban.Syntax.ScriptBinaryExpression.Evaluate(TemplateContext context, SourceSpan span, ScriptBinaryOperator op, SourceSpan leftSpan, Object leftValue, SourceSpan rightSpan, Object rightValue)
   at Scriban.Syntax.ScriptBinaryExpression.Evaluate(TemplateContext context)
   at Scriban.TemplateContext.Evaluate(ScriptNode scriptNode, Boolean aliasReturnedFunction)
   at Scriban.Syntax.ScriptBinaryExpression.Evaluate(TemplateContext context)
   at Scriban.TemplateContext.Evaluate(ScriptNode scriptNode, Boolean aliasReturnedFunction)
   at Scriban.Syntax.ScriptBinaryExpression.Evaluate(TemplateContext context)
   at Scriban.TemplateContext.Evaluate(ScriptNode scriptNode, Boolean aliasReturnedFunction)
   at Scriban.Syntax.ScriptBinaryExpression.Evaluate(TemplateContext context)
   at Scriban.TemplateContext.Evaluate(ScriptNode scriptNode, Boolean aliasReturnedFunction)
   at Scriban.Syntax.ScriptIfStatement.Evaluate(TemplateContext context)
   at Scriban.TemplateContext.Evaluate(ScriptNode scriptNode, Boolean aliasReturnedFunction)
   at Scriban.Syntax.ScriptBlockStatement.Evaluate(TemplateContext context)
   at Scriban.TemplateContext.Evaluate(ScriptNode scriptNode, Boolean aliasReturnedFunction)
   at Scriban.Syntax.ScriptFunction.Invoke(TemplateContext context, ScriptNode callerContext, ScriptArray arguments, ScriptBlockStatement blockStatement)
   at Scriban.Syntax.ScriptFunctionCall.Call(TemplateContext context, ScriptNode callerContext, Object functionObject, Boolean processPipeArguments, IReadOnlyList`1 arguments)
   at Scriban.Syntax.ScriptFunctionCall.Evaluate(TemplateContext context)
   at Scriban.TemplateContext.Evaluate(ScriptNode scriptNode, Boolean aliasReturnedFunction)
   at Scriban.Syntax.ScriptPipeCall.Evaluate(TemplateContext context)
   at Scriban.TemplateContext.Evaluate(ScriptNode scriptNode, Boolean aliasReturnedFunction)
   at Scriban.Syntax.ScriptExpressionStatement.Evaluate(TemplateContext context)
   at Scriban.TemplateContext.Evaluate(ScriptNode scriptNode, Boolean aliasReturnedFunction)
   at Scriban.Syntax.ScriptBlockStatement.Evaluate(TemplateContext context)
   at Scriban.TemplateContext.Evaluate(ScriptNode scriptNode, Boolean aliasReturnedFunction)
   at Scriban.Syntax.ScriptFunction.Invoke(TemplateContext context, ScriptNode callerContext, ScriptArray arguments, ScriptBlockStatement blockStatement)
   at Scriban.Syntax.ScriptFunctionCall.Call(TemplateContext context, ScriptNode callerContext, Object functionObject, Boolean processPipeArguments, IReadOnlyList`1 arguments)
   at Scriban.Syntax.ScriptFunctionCall.Evaluate(TemplateContext context)
   at Scriban.TemplateContext.Evaluate(ScriptNode scriptNode, Boolean aliasReturnedFunction)
   at Scriban.Syntax.ScriptExpressionStatement.Evaluate(TemplateContext context)
   at Scriban.TemplateContext.Evaluate(ScriptNode scriptNode, Boolean aliasReturnedFunction)
   at Scriban.Syntax.ScriptBlockStatement.Evaluate(TemplateContext context)
   at Scriban.TemplateContext.Evaluate(ScriptNode scriptNode, Boolean aliasReturnedFunction)
   at Scriban.Syntax.ScriptForStatement.LoopItem(TemplateContext context, LoopState state)
   at Scriban.Syntax.ScriptForStatement.EvaluateImpl(TemplateContext context)
   at Scriban.Syntax.ScriptLoopStatementBase.Evaluate(TemplateContext context)
   at Scriban.TemplateContext.Evaluate(ScriptNode scriptNode, Boolean aliasReturnedFunction)
   at Scriban.Syntax.ScriptBlockStatement.Evaluate(TemplateContext context)
   at Scriban.TemplateContext.Evaluate(ScriptNode scriptNode, Boolean aliasReturnedFunction)
   at Scriban.Syntax.ScriptPage.Evaluate(TemplateContext context)
   at Scriban.TemplateContext.Evaluate(ScriptNode scriptNode, Boolean aliasReturnedFunction)
   at Scriban.Template.EvaluateAndRender(TemplateContext context, Boolean render)
   at Scriban.Template.Render(TemplateContext context)
```

Looks like it's trying to coerce the `ulong` down to the `int`, rather than trying to coerce the `int` up to the `ulong`

**Comment by xoofx** [comment]
I don't remember the details, but ulong might not be well supported (maybe long was only implemented). It was new for Scriban 4.0 for [kalk](https://kalk.dev) but tests were not all implemented for the new introduced features.

**Comment by xoofx** [comment]
Fixed by commit 93fef08
